Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+                        "Editor Mode Feedback System", "Manages discovery of Editr Mode Feedback effects")
+                        ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("System", 0xc94d118b))
+                        ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                        ;
+                }
+            }
+        }
+
+        EditorEditorModeFeedbackSystemComponent::~EditorEditorModeFeedbackSystemComponent() = default;
+
+        void EditorEditorModeFeedbackSystemComponent::Activate()
+        {
+            AzToolsFramework::Components::EditorComponentBase::Activate();
+            AzToolsFramework::ViewportEditorModeNotificationsBus::Handler::BusConnect(AzToolsFramework::GetEntityContextId());
+            AZ::Interface<EditorModeFeedbackInterface>::Register(this);
+            AZ::TickBus::Handler::BusConnect();
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::Deactivate()
+        {
+            AZ::TickBus::Handler::BusDisconnect();
+            AZ::Interface<EditorModeFeedbackInterface>::Unregister(this);
+            AzToolsFramework::ViewportEditorModeNotificationsBus::Handler::BusDisconnect();
+            AzToolsFramework::Components::EditorComponentBase::Deactivate();
+        }
+
+        bool EditorEditorModeFeedbackSystemComponent::IsEnabled() const
+        {
+            return m_enabled;
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::RegisterDrawableComponent(
+            EntityComponentIdPair entityComponentId, const MeshFeatureProcessorInterface::MeshHandle& meshHandle)
+        {
+            // Overwrite any existing mesh handle for this component-entity id
+            auto& meshHandleDrawPackets =
+                m_entityComponentMeshHandleDrawPackets[entityComponentId.GetEntityId()][entityComponentId.GetComponentId()];
+            meshHandleDrawPackets.m_meshHandle = &meshHandle;
+
+            // The same component can call RegisterDrawableEntity multiple times in order to update its model asset so always
+            // clear any existing draw packets for the component upon registration
+            meshHandleDrawPackets.m_meshDrawPackets.clear();
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnEditorModeActivated(
+            [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState,
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            // Purge the draw packets for all registered 
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                for (auto& [entityId, componentMeshHandleDrawPackets] : m_entityComponentMeshHandleDrawPackets)
+                {
+                    for (auto& [componentid, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
```

Comment:
question: Will `[[maybe_unused]]` be needed here? (might not be necessary with the structured binding)

Reply:
This doesn't appear to be necessary for structured bindings 👍